### DISCUSSION TITLE
Question: Which fields in `pyproject.toml` are mandatory, which will stay optional

### DISCUSSION THREAD

**Discussion by schwaerz** [pdm-project/pdm#2933] [Q&A] [answered]
## Is your feature/enhancement proposal related to a problem? Please describe.

We are about to decide to using pdm for managing our python environment.
One of the reasons to do so was that pdm does not require us to specify name, version, description, ... in `pyproject.toml`. For Conan projects, those details will be already in the `conanfile.py`.

Today the question came up whether we can rely on not having to specify this metatata if not required - or wether this soon will possibly change, esp since [PEP 621](https://peps.python.org/pep-0621/#backwards-compatibility) says about `[project.name]`:
> Tools MUST require users to statically define this field.

I tried to find some more information on that in the documentation and open and closed tickets. No luck so far.

## Describe the solution you'd like

Would like to have clarity about what we can possibly expect from pdm in future wrt. to the mandatory fields to be defined in `pyproject.toml`.

Currently for Conan projects we stick to the minimum, which is
* `[project.requires_python]`
* `[project.dependencies]`

And we'd like to have some clue whether there are any plans to change that in future.

**Comment by frostming** [comment, marked as answer]
According to PEP 621 only `project.name` is mandatory. PDM won't enforce other fields beyond the standard.

**Reply by schwaerz** [reply]
@frostming Thanks for the quick answer. Luckily `project.name` is currently not enforced, either - and we are hoping it stays like this...
